array/array_test.go

```diff
@@ -21,7 +21,7 @@ func TestString(t *testing.T) {
 					b.Append("a")
 				}
 			},
-			sz: 256,
+			sz: 0,
```

Comment:
A quick read of this test indicates that we're changing the behavior of this api, and so the test needs updating. Is that right?

Reply:
Yes. I wrote the test before implementing this. Now that the builder produces the more optimized format, and this optimized format doesn't use arrow buffers at all, it now "allocates nothing". That's not strictly true but that's true for the sake of this.